adal/src/main/java/com/microsoft/aad/adal/AuthenticationException.java

```diff
@@ -70,6 +84,89 @@ public AuthenticationException(ADALError code, String details) {
     public AuthenticationException(ADALError code, String details, Throwable throwable) {
         super(details, throwable);
         mCode = code;
+
+        if (null == throwable) {
+            return;
+        }
+
+        if (throwable instanceof AuthenticationException) {
+            mServiceStatusCode = ((AuthenticationException) throwable).getServiceStatusCode();
+            mResponseBody = ((AuthenticationException) throwable).getResponseBody();
+            mHttpResponseHeaders = ((AuthenticationException) throwable).getHttpResponseHeaders();
```

Comment:
nit: if its more than once, should we create a variable that casts it to AuthenticationException